src/pudl/transform/eia861.py

```diff
@@ -723,13 +717,39 @@ def _tidy_class_dfs(
     data_cols = data_cols.stack(level=0, dropna=False).reset_index()
     denorm_cols = _filter_non_class_cols(raw_df, class_list).reset_index()
 
+    # Check to make sure that the idx_cols are actually valid primary key cols:
+    # This is tricky, because NA values in the BA Code column creates actual duplicate
+    # values. These NA values are filled with UNK and the duplicates are later dropped,
+    # which means we are losing data. But it's not obvious how we could actually fill
+    # in real BA codes, so we consolidate the indistinguisbale records.
+    # These are only a fraction of a percent of all records, and only affect a couple
+    # of tables. In sales_eia861 there are 67 duplicate records.
+    # See: https://github.com/catalyst-cooperative/pudl/issues/2638
+    dupe_mask = data_cols.duplicated(subset=idx_cols + [class_type], keep=False)
+    dupes = data_cols[dupe_mask]
+    fraction_dupes = len(dupes) / len(data_cols)
+    err_msg = (
+        f"{df_name} table: Found {len(dupes)}/{len(data_cols)} "
+        f"({fraction_dupes:0.2%}) records with duplicated PKs. "
+    )
+    if fraction_dupes <= 0.005:
```

Comment:
Do you want this to be a configurable percentage that we call in through `context`? Either way seems ok to me, we just often do so rather than hardcoding this value. The value itself seems reasonable though!

Reply:
Hmm, I don't think I've really used the dagster context this way before. Is there an example you're thinking of?  I know magic numbers aren't great. I just wanted something that would alert us if the current very small number of duplicates were to change.

Reply:
I'm thinking of `thresh` in `denorm_fuel_by_plant_ferc1` but I think in this case either way is alright and this seems like a reasonable value to hard-code in.